tests/testthat/test-log-likelihood-linear.R

```diff
@@ -0,0 +1,18 @@
+test_that("log_likelihood_linear_gradient against numerical gradient", {
+
+  n_obs <- 32; n_pred <- 4
+  data <- simulate_data(n_obs, n_pred, model = 'linear', seed = 2883)
```

Comment:
Don't mix single and double quotes. Tidyverse style guide recommends double quotes. Same elsewhere in code.